### PR TITLE
Update Documentation 

### PR DESCRIPTION
**Pull request by dgil**
I'm in the process of updating this workshop.

The goal of this PR is:
* Update the documentation to be up-to-date with the current status of Docker
* Add `docker-compose` documentation
* Add `docker-machine` documentation

### CODE REVIEW

File: doc/01-docker-machine/index.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+```
+docker-machine create -d virtualbox default
+```
+
+Start a VM named `default`:
+
+```
+docker-machine start default
+```
+
```
Comment (spiddy) [inline doc/01-docker-machine/index.md]: Ok, now that we have docker-machine, maybe add an example what we can do with it (`docker run hello-world`) before stopping it.

Comment (spiddy) [inline doc/01-docker-machine/index.md]: And to do that you'll need to `eval $(docker machine env default)`

File: doc/02-docker-compose/index.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+We've already created a simple app in `code/docker-node` that uses node.js with express and redis.
+
+1. Go to `code/docker-node` folder
+
+2. Review `Dockerfile`:
+
+```
+FROM mhart/alpine-node:latest
+RUN mkdir /code
```
Comment (spiddy) [inline doc/02-docker-compose/index.md]: mkdir is not necessary, ADD will create it.

Comment (dgil) [inline doc/02-docker-compose/index.md]: @spiddy I think is necessary for the `workdir`

**Comment by spiddy** [comment]
LGTM now.

I applied some necessary changes.

**Pull request merged**